### PR TITLE
chore: adding type Literal to constants for files in appsec

### PR DESCRIPTION
**Pull request by rachelyangdog**
Any constant can be typed as being a literal instead of a type. This change will ensure that anywhere we use a constant for typing it means literally the value instead of a type of str. In order for this to work for Python < 3.8 we will need to add typing_extensions as a dependency.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/appsec/_constants.py [hunk 1]
```diff
@@ -41,85 +47,103 @@ def __getitem__(self, k: str) -> Any:
...
+    TELEMETRY_REPORT_LVL: Literal["DD_IAST_TELEMETRY_VERBOSITY"] = "DD_IAST_TELEMETRY_VERBOSITY"
+    LAZY_TAINT: Literal["_DD_IAST_LAZY_TAINT"] = "_DD_IAST_LAZY_TAINT"
+    JSON: Literal["_dd.iast.json"] = "_dd.iast.json"
+    ENABLED: Literal["_dd.iast.enabled"] = "_dd.iast.enabled"
+    CONTEXT_KEY: Literal["_iast_data"] = "_iast_data"
+    PATCH_MODULES: Literal["_DD_IAST_PATCH_MODULES"] = "_DD_IAST_PATCH_MODULES"
+    DENY_MODULES: Literal["_DD_IAST_DENY_MODULES"] = "_DD_IAST_DENY_MODULES"
+    SEP_MODULES: Literal[","] = ","
+    REQUEST_IAST_ENABLED: Literal["_dd.iast.request_enabled"] = "_dd.iast.request_enabled"
+    TEXT_TYPES: Literal["str", "bytes", "bytearray"] = (str, bytes, bytearray)
```
Comment (ZStriker19) [inline ddtrace/appsec/_constants.py:121]: This type needs to be changed to a Literal tuple that contains those those string values.

Comment (ZStriker19) [reply ddtrace/appsec/_constants.py:121]: See: https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/66375/workflows/30c58678-7352-4dcf-a1db-1b2efbfb6fc0/jobs/4079892

File: ddtrace/appsec/_handlers.py [hunk 2]
```diff
@@ -28,6 +28,7 @@
 
 
 log = get_logger(__name__)
+
```
Comment (ZStriker19) [inline ddtrace/appsec/_handlers.py:31]:
```suggestion
```
let's remove this

File: ddtrace/appsec/_constants.py [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
+if sys.version_info >= (3, 8):
+    from typing import Literal  # noqa:F401
+else:
+    from typing_extensions import Literal  # noqa:F401
+
 from typing import Any
 from typing import Iterator
 
 from ddtrace.internal.constants import HTTP_REQUEST_BLOCKED
-from ddtrace.internal.constants import REQUEST_PATH_PARAMS
```
Comment (ZStriker19) [inline ddtrace/appsec/_constants.py:6]: why do we remove these imports?

Comment (ZStriker19) [reply ddtrace/appsec/_constants.py:6]: Ah to make them literals. Let's just keep these as they are, with the imports, and then when you change ddtrace.internal.constants, the globals here will be changed as well.

**Review by gnufede — APPROVED** [review]

File: ddtrace/appsec/_constants.py [hunk 4]
```diff
@@ -260,20 +288,21 @@ class DEFAULT(metaclass=Constant_Class):
...
-    STACK_TRACES = "_dd.stack"
-    STACK_TRACE_ID = "stack_id"
-    EP_ENABLED = "DD_APPSEC_RASP_ENABLED"
-    STACK_TRACE_ENABLED = "DD_APPSEC_STACK_TRACE_ENABLED"
-    MAX_STACK_TRACES = "DD_APPSEC_MAX_STACK_TRACES"
-    MAX_STACK_TRACE_DEPTH = "DD_APPSEC_MAX_STACK_TRACE_DEPTH"
+    STACK_TRACES: Literal["_dd.stack"] = "_dd.stack"
+    STACK_TRACE_ID: Literal["stack_id"] = "stack_id"
+    EP_ENABLED: Literal["DD_APPSEC_RASP_ENABLED"] = "DD_APPSEC_RASP_ENABLED"
+    # continue for the rest of the constants
```
Comment (gnufede) [inline ddtrace/appsec/_constants.py:294]: This comment doesn't add value
```suggestion
```

File: ddtrace/appsec/_constants.py [hunk 5]
```diff
@@ -138,73 +164,75 @@ class WAF_DATA_NAMES(metaclass=Constant_Class):
...
     """string names used by the library for tagging data from requests in context or span"""
 
-    REQUEST_BODY = "http.request.body"
-    REQUEST_QUERY = "http.request.query"
-    REQUEST_HEADERS_NO_COOKIES = "http.request.headers"
-    REQUEST_HEADERS_NO_COOKIES_CASE = "http.request.headers_case_sensitive"
-    REQUEST_URI_RAW = "http.request.uri"
-    REQUEST_ROUTE = "http.request.route"
-    REQUEST_METHOD = "http.request.method"
-    REQUEST_PATH_PARAMS = REQUEST_PATH_PARAMS
```
Comment (ZStriker19) [inline ddtrace/appsec/_constants.py:158]: Go back to this

**Pull request merged**